packages/dds/tree/docs/storage/compatibility.md

```diff
@@ -0,0 +1,247 @@
+These tests generally won't be necessary to modify when adding new formats or persisted configuration.
+
+### Configuration-specific Upgrade Tests
+
+The persisted configuration store tests and its integration tests into SharedTree will verify that the scheme for upgrading
+configuration generally works correctly, but they can't generically exercise logic specific to some particular format change.
+Any format change which is more involved than an encoding change should have some targeted tests which verify interesting aspects of the upgrade process work correctly.
+For example, legacy SharedTree's introduction of IdCompressor involved [testing the upgrade process](../../../../../experimental/dds/tree/src/test/utilities/SharedTreeVersioningTests.ts) and specifically the synchronization point logic (see 'generates unique IDs after upgrading from 0.0.2').
+
+These tests should cover logic invoked as part of `onProtocolChange` and `reSubmitPendingOps`.
+
+### Snapshot Tests
+
+The last dimension of compatibility concerns direct or indirect collaboration between clients using different versions of SharedTree source code.
+This is a vast area that could use more well-established framework testing support, but snapshot testing is a relatively effective category for
+catching regressions.
+
+The idea behind snapshot testing is to verify a document produced using one version of the code is still usable using another version of the code.
+It's typically implemented by source-controlling a set of serialized test documents and writing a suite which verifies:
+
+1. The current version of the code serializes the documents in the same way
+1. The current version of the code is capable of loading those documents
+
+A few examples (which may be exhaustive) of snapshot tests are:
+
+-   [Legacy SharedTree](../../../../../experimental/dds/tree/src/test/Summary.tests.ts)
+-   [Sequence / SharedString](../../../sequence/src/test/snapshotVersion.spec.ts)
+-   [e2e Snapshot tests](../../../../test/snapshots/README.md)
+
+The first two examples generate their documents in a similar fashion: in addition to source-controlling the serialized snapshot,
+they have functions to "re-build" the same state from scratch using the current version of the code.
+The e2e snapshot tests accomplish this by serializing the op stream alongside the snapshots.
+
+This enables checking a third invariant: that loading a document serialized using some previous version of the code and
+generating it from scratch using the current version of the code produce the same state.
+
+Snapshot tests are effective at catching changes which inadvertently modify document format over time.
+SharedTree should have snapshot testing for each of its supported configurations.
+
+## Current State
+
+Since a DDS has effectively 2 ways of creating persisted state, one can audit both sources to find all code that impacts persisted state as follows:
```

Comment:
```suggestion
Since a DDS has effectively 2 ways of creating persisted state (summaries and ops), one can audit both sources to find all code that impacts persisted state as follows:
```